src/js/zqd/zqd.ts

```diff
@@ -150,9 +150,32 @@ export class ZQD {
     return "localhost:9867"
   }
 
-  close() {
+  async close(): Promise<void> {
     if (this.zqd) {
+      // give zqd 5 seconds to exit
+      let giveUp = false
+      setTimeout(() => {
+        giveUp = true
+      }, 5000)
+
       this.zqd.kill("SIGTERM")
+      while (!giveUp && (await isUp())) {
+        await new Promise((res) => setTimeout(res, 500))
+      }
+
+      if (giveUp) {
+        throw new Error("gave up waiting for zqd to shutdown")
+      }
```

Comment:
I think it should just log what happened here.

If giveUp, log.error("zqd failed to shutdown in time")
else log.info("zqd has shutdown")